### PR TITLE
t/6: Added Controller.addCollection() method

### LINKED ISSUE #6: Add a shorthand method for registering a controller collection
Moved from https://github.com/ckeditor/ckeditor5-core/issues/214

```
this.collections.add( new ControllerCollection( 'buttons' ) );
```

into:

```
this.addCollection( 'buttons' );
```

### PR DESCRIPTION
**Pull request by oleq**
Closes #6. Requires #34.

### CODE REVIEW

File: src/controller.js [hunk 1]
```diff
@@ -153,6 +154,21 @@ export default class Controller {
...
 
 	/**
+	 * Adds a new collection to {@link ui.Controller#collections}.
+	 *
+	 * @param {String} collectionName Name of the controller collection.
+	 * @param {utils.Locale} [locale] The {@link ckeditor5.Editor#locale editor's locale} instance.
+	 * See {@link ui.ControllerCollection#locale}.
+	 */
+	addCollection( collectionName, locale ) {
+		const collection = new ControllerCollection( collectionName, locale );
```
Comment (Reinmar) [inline src/controller.js]: Do we have to pass the locale? Can't it be taken here from `this.view.locale`?

File: src/controller.js [hunk 2]
```diff
@@ -153,6 +153,22 @@ export default class Controller {
...
 
 	/**
+	 * Adds a new collection to {@link ui.Controller#collections}.
+	 *
+	 * @param {String} collectionName Name of the controller collection.
+	 * @param {utils.Locale} [locale] The {@link ckeditor5.Editor#locale editor's locale} instance.
+	 * See {@link ui.ControllerCollection#locale}.
+	 * @returns {ui.ControllerCollection} The new collection instance.
+	 */
+	addCollection( collectionName, locale ) {
```
Comment (Reinmar) [inline src/controller.js]: My previous comment disappeared – can't we take `locale` from `this.view.locale`?

**Comment by Reinmar** [comment]
If https://github.com/ckeditor/ckeditor5-ui/pull/50/files#r72650075 can be simplified, feel free to create an issue for it. I'm merging the rest of the code.

File: src/controller.js (hunk 2 above)
Comment (oleq) [inline src/controller.js]: That's a good idea. I'll fix it post–review.

**Pull request merged**